<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2021-09-28" />

<title>Methylation profiling of paediatric Bronchoalveolar Lavage (BAL)</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-BAL-meth-ref</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Methylation profiling of paediatric Bronchoalveolar Lavage (BAL)</h1>
<h3 class="subtitle">Establishing a methylation reference panel of BAL consitutent cell types</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">2021-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-BAL-meth-ref/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210927code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210927code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210927)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedbalmethreftreee968190c85470b191cab0de6adabaa18a60ace5ctargetblanke968190a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/tree/e968190c85470b191cab0de6adabaa18a60ace5c" target="_blank">e968190</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedbalmethreftreee968190c85470b191cab0de6adabaa18a60ace5ctargetblanke968190a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/tree/e968190c85470b191cab0de6adabaa18a60ace5c" target="_blank">e968190</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/deident.R
    Ignored:    data/idat/

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/dataPreprocess.Rmd</code>) and HTML (<code>docs/dataPreprocess.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/e968190c85470b191cab0de6adabaa18a60ace5c/analysis/dataPreprocess.Rmd" target="_blank">e968190</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-28
</td>
<td>
wflow_publish(c(“analysis/dataPreprocess.Rmd”, “analysis/estimateCellProportions.Rmd”))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/dataPreprocess.html" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/1114da3d98ebc1f90be57f2626e71b16d7ea20f7/analysis/dataPreprocess.Rmd" target="_blank">1114da3</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/dataPreprocess.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-import" class="section level1">
<h1>Data import</h1>
<p>Load all the packages required for analysis.</p>
<pre class="r"><code>library(here)
library(workflowr)
library(limma)
library(minfi)
library(missMethyl)
library(matrixStats)
library(minfiData)
library(stringr)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(IlluminaHumanMethylationEPICmanifest)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(glue)</code></pre>
<p>Load the EPIC array annotation data that describes the genomic context of each of the probes on the array.</p>
<pre class="r"><code># Get the EPICarray annotation data
annEPIC &lt;- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
annEPIC %&gt;% data.frame %&gt;%
  dplyr::select(&quot;chr&quot;,
                &quot;pos&quot;,
                &quot;strand&quot;,
                &quot;UCSC_RefGene_Name&quot;,
                &quot;UCSC_RefGene_Group&quot;) %&gt;%
  head() %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left"></th>
<th align="left">chr</th>
<th align="right">pos</th>
<th align="left">strand</th>
<th align="left">UCSC_RefGene_Name</th>
<th align="left">UCSC_RefGene_Group</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cg18478105</td>
<td align="left">chr20</td>
<td align="right">61847650</td>
<td align="left">-</td>
<td align="left">YTHDF1</td>
<td align="left">TSS200</td>
</tr>
<tr class="even">
<td align="left">cg09835024</td>
<td align="left">chrX</td>
<td align="right">24072640</td>
<td align="left">-</td>
<td align="left">EIF2S3</td>
<td align="left">TSS1500</td>
</tr>
<tr class="odd">
<td align="left">cg14361672</td>
<td align="left">chr9</td>
<td align="right">131463936</td>
<td align="left">+</td>
<td align="left">PKN3</td>
<td align="left">TSS1500</td>
</tr>
<tr class="even">
<td align="left">cg01763666</td>
<td align="left">chr17</td>
<td align="right">80159506</td>
<td align="left">+</td>
<td align="left">CCDC57</td>
<td align="left">Body</td>
</tr>
<tr class="odd">
<td align="left">cg12950382</td>
<td align="left">chr14</td>
<td align="right">105176736</td>
<td align="left">+</td>
<td align="left">INF2;INF2</td>
<td align="left">Body;Body</td>
</tr>
<tr class="even">
<td align="left">cg02115394</td>
<td align="left">chr13</td>
<td align="right">115000168</td>
<td align="left">+</td>
<td align="left">CDC16;CDC16</td>
<td align="left">TSS200;TSS200</td>
</tr>
</tbody>
</table>
<p>Read the sample information and IDAT file paths into R.</p>
<pre class="r"><code># absolute path to the directory where the data is (relative to the Rstudio project)
dataDirectory &lt;- here(&quot;data&quot;)

# read in the sample sheet for the experiment
read.metharray.sheet(dataDirectory, 
                     pattern = &quot;Samplesheet_BAL_reference.csv&quot;) %&gt;%
  mutate(Basename = gsub(&quot;c(\&quot;&quot;,&quot;&quot;, Basename, fixed=TRUE)) %&gt;%
  mutate(Sample_ID = paste(Slide, Array, sep = &quot;_&quot;)) %&gt;%
  mutate_if(is.character, stringr::str_replace_all, pattern = &quot;Granuloycte&quot;, 
            replacement = &quot;Granulocyte&quot;) %&gt;%
  mutate_if(is.character, stringr::str_replace_all, pattern = &quot;Epithelialcell&quot;, 
            replacement = &quot;EpithelialCell&quot;) %&gt;%
  mutate_if(is.character, stringr::str_replace_all, pattern = &quot;Old&quot;, 
            replacement = &quot;Erasmus MC&quot;) %&gt;%
  mutate_if(is.character, stringr::str_replace_all, pattern = &quot;New&quot;, 
            replacement = &quot;GenomeScan&quot;) %&gt;%
  dplyr::select(Sample_ID, Sample_Group, Sample_source, Sample_run, 
                Basename) -&gt; targets</code></pre>
<pre><code>[1] &quot;/oshlack_lab/jovana.maksimovic/projects/MCRI/shivanthan.shanthikumar/paed-BAL-meth-ref/data/Samplesheet_BAL_reference.csv&quot;</code></pre>
<pre class="r"><code>targets %&gt;% dplyr::select(-Basename) %&gt;% knitr::kable()</code></pre>
<table>
<thead>
<tr class="header">
<th align="left">Sample_ID</th>
<th align="left">Sample_Group</th>
<th align="left">Sample_source</th>
<th align="left">Sample_run</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">202905570075_R02C01</td>
<td align="left">EpithelialCell</td>
<td align="left">A1</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="even">
<td align="left">202900540047_R01C01</td>
<td align="left">Macrophage</td>
<td align="left">M1</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="odd">
<td align="left">202900540047_R02C01</td>
<td align="left">Macrophage</td>
<td align="left">M2</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="even">
<td align="left">202900540047_R03C01</td>
<td align="left">Macrophage</td>
<td align="left">M3</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="odd">
<td align="left">202900540047_R04C01</td>
<td align="left">Granulocyte</td>
<td align="left">G1</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="even">
<td align="left">202900540047_R05C01</td>
<td align="left">Granulocyte</td>
<td align="left">G2</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="odd">
<td align="left">202900540047_R06C01</td>
<td align="left">Granulocyte</td>
<td align="left">G3</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="even">
<td align="left">202900540047_R07C01</td>
<td align="left">Lymphocyte</td>
<td align="left">L1</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="odd">
<td align="left">202900540047_R08C01</td>
<td align="left">Lymphocyte</td>
<td align="left">L2</td>
<td align="left">Erasmus MC</td>
</tr>
<tr class="even">
<td align="left">204071680134_R04C01</td>
<td align="left">Macrophage</td>
<td align="left">M1</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="odd">
<td align="left">204071680134_R05C01</td>
<td align="left">Macrophage</td>
<td align="left">M2</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="even">
<td align="left">204071680134_R06C01</td>
<td align="left">Granulocyte</td>
<td align="left">G1</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="odd">
<td align="left">204071680134_R07C01</td>
<td align="left">Lymphocyte</td>
<td align="left">L1</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="even">
<td align="left">204071680134_R08C01</td>
<td align="left">EpithelialCell</td>
<td align="left">AEC1</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="odd">
<td align="left">204074230020_R01C01</td>
<td align="left">Raw</td>
<td align="left">CF4</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="even">
<td align="left">204074230020_R02C01</td>
<td align="left">Raw</td>
<td align="left">CF1</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="odd">
<td align="left">204074230020_R03C01</td>
<td align="left">Raw</td>
<td align="left">CF3</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="even">
<td align="left">204074230020_R04C01</td>
<td align="left">Raw</td>
<td align="left">CF2</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="odd">
<td align="left">204074230020_R05C01</td>
<td align="left">Raw</td>
<td align="left">CF5</td>
<td align="left">GenomeScan</td>
</tr>
<tr class="even">
<td align="left">204074230020_R06C01</td>
<td align="left">Raw</td>
<td align="left">CTRL</td>
<td align="left">GenomeScan</td>
</tr>
</tbody>
</table>
<p>Read in the raw methylation data.</p>
<pre class="r"><code># read in the raw data from the IDAT files
rgSet &lt;- read.metharray.exp(targets = targets)</code></pre>
<pre><code>Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls

Warning in readChar(con, nchars = n): truncating string with embedded nuls</code></pre>
<pre class="r"><code>rgSet</code></pre>
<pre><code>class: RGChannelSet 
dim: 1051815 20 
metadata(0):
assays(2): Green Red
rownames(1051815): 1600101 1600111 ... 99810990 99810992
rowData names(0):
colnames(20): 202905570075_R02C01 202900540047_R01C01 ...
  204074230020_R05C01 204074230020_R06C01
colData names(6): Sample_ID Sample_Group ... Basename filenames
Annotation
  array: IlluminaHumanMethylationEPIC
  annotation: ilm10b4.hg19</code></pre>
</div>
<div id="quality-control" class="section level1">
<h1>Quality control</h1>
<p>Calculate the detection P-values for each probe so that we can check for any failed samples.</p>
<pre class="r"><code># calculate the detection p-values
detP &lt;- detectionP(rgSet)
head(detP[, 1:5]) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">202905570075_R02C01</th>
<th align="right">202900540047_R01C01</th>
<th align="right">202900540047_R02C01</th>
<th align="right">202900540047_R03C01</th>
<th align="right">202900540047_R04C01</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cg18478105</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">cg09835024</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">cg14361672</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">cg01763666</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">cg12950382</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">cg02115394</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
<p>We can see the mean detection p-values across all the samples for the two runs.</p>
<pre class="r"><code>dat &lt;- reshape2::melt(colMeans(detP))
dat$type &lt;- targets$Sample_Group
dat$sample &lt;- targets$Sample_source
dat$run &lt;- targets$Sample_run

pal &lt;- scales::hue_pal()(length(unique(targets$Sample_Group)))
names(pal) &lt;- c(&quot;Granulocyte&quot;,&quot;Raw&quot;,&quot;Lymphocyte&quot;,
                &quot;Macrophage&quot;,&quot;EpithelialCell&quot;)
  
# examine mean detection p-values across all samples to identify failed samples
p &lt;- ggplot(dat, aes(x = sample, y = value, fill = type)) +
  geom_bar(stat = &quot;identity&quot;) +
  facet_wrap(vars(run), ncol = 2, scales = &quot;free_x&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Detection p-value&quot;, fill = &quot;Sample type&quot;) +
  theme(axis.text.y = element_text(size = 7),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = &quot;bottom&quot;) +
  scale_fill_manual(values = pal)
p</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-6-1">
Past versions of unnamed-chunk-6-1.png
</button>
</p>
<div id="fig-unnamed-chunk-6-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-6-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="sample-filtering" class="section level2">
<h2>Sample filtering</h2>
<p>Filter out samples with mean detection p-values &gt; 0.01 to avoid filtering out too many probes downstream.</p>
<pre class="r"><code>keep &lt;- colMeans(detP) &lt; 0.01
table(keep)</code></pre>
<pre><code>keep
TRUE 
  20 </code></pre>
<pre class="r"><code>rgSet &lt;- rgSet[, keep]
targets &lt;- targets[keep, ]
detP &lt;- detP[, keep]</code></pre>
</div>
<div id="normalisation" class="section level2">
<h2>Normalisation</h2>
<p>Normalise the data.</p>
<pre class="r"><code># normalize the data; this results in a GenomicRatioSet object
mSetSq &lt;- preprocessQuantile(rgSet)</code></pre>
<pre><code>[preprocessQuantile] Mapping to genome.</code></pre>
<pre><code>[preprocessQuantile] Fixing outliers.</code></pre>
<pre><code>[preprocessQuantile] Quantile normalizing.</code></pre>
<pre class="r"><code>mSetSq</code></pre>
<pre><code>class: GenomicRatioSet 
dim: 865859 20 
metadata(0):
assays(2): M CN
rownames(865859): cg14817997 cg26928153 ... cg07587934 cg16855331
rowData names(0):
colnames(20): 202905570075_R02C01 202900540047_R01C01 ...
  204074230020_R05C01 204074230020_R06C01
colData names(9): Sample_ID Sample_Group ... yMed predictedSex
Annotation
  array: IlluminaHumanMethylationEPIC
  annotation: ilm10b4.hg19
Preprocessing
  Method: Raw (no normalization or bg correction)
  minfi version: 1.36.0
  Manifest version: 0.3.0</code></pre>
<pre class="r"><code># create a MethylSet object from the raw data for plotting
mSetRaw &lt;- preprocessRaw(rgSet)</code></pre>
<pre class="r"><code>bSq &lt;- getBeta(mSetSq)
bRaw &lt;- getBeta(mSetRaw)

sq &lt;- reshape2::melt(bSq)
sq$type &lt;- targets$Sample_Group
sq$process &lt;- &quot;Normalised&quot;

raw &lt;- reshape2::melt(bRaw)
raw$type &lt;- targets$Sample_Group
raw$process &lt;- &quot;Raw&quot;

dat &lt;- bind_rows(sq, raw)

colnames(dat)[1:3] &lt;- c(&quot;cpg&quot;, &quot;ID&quot;, &quot;beta&quot;)

ggplot(data = dat, 
       aes(x = beta, colour = type)) +
  geom_density(show.legend = NA) +
  labs(colour = &quot;Sample type&quot;) +
  facet_wrap(vars(process), ncol = 2) +
  scale_color_manual(values = pal)</code></pre>
<pre><code>Warning: Removed 270 rows containing non-finite values (stat_density).</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-9-1">
Past versions of unnamed-chunk-9-1.png
</button>
</p>
<div id="fig-unnamed-chunk-9-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-9-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="data-exploration-before-probe-filtering" class="section level2">
<h2>Data exploration (before probe filtering)</h2>
<p>Explore the data to look for any structure. As expected, sex is the most significant source of variation. The patient samples, in particular, are clearly grouping by sex.</p>
<pre class="r"><code>mDat &lt;- getM(mSetSq)
mds &lt;- plotMDS(mDat, top = 1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group, 
              run = targets$Sample_run,
              source = targets$Sample_source,
              sex = targets$Sex)

p1 &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
       x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;) +
  ggtitle(&quot;All data&quot;) +
  scale_color_manual(values = pal)

p1</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-10-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Colour ONLY the sorted cells samples using different variables. Encouragingly, the cell types cluster tightly when samples from both runs are combined, indicating that differences between cell types are more significant than any bath effects between the two runs. The cell types also cluster as expected for each of the individual runs.</p>
<pre class="r"><code>cells &lt;- !(targets$Sample_Group %in% c(&quot;Raw&quot;))
mds &lt;- plotMDS(mDat[, cells], top = 1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group[cells], 
              run = targets$Sample_run[cells],
              source = targets$Sample_source[cells])

p1 &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
       x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;) +
  ggtitle(&quot;Cell types&quot;) +
  scale_color_manual(values = pal[-2])

p1</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-11-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="probe-filtering" class="section level2">
<h2>Probe filtering</h2>
<p>Filter out poor performing probes, sex chromosome probes, SNP probes and cross reactive probes.</p>
<pre class="r"><code># ensure probes are in the same order in the mSetSq and detP objects
detP &lt;- detP[match(featureNames(mSetSq), rownames(detP)),]
# remove any probes that have failed in one or more samples
keep &lt;- rowSums(detP &lt; 0.01) == ncol(mSetSq)
# subset data
mSetSqFlt &lt;- mSetSq[keep,]
mSetSqFlt</code></pre>
<pre><code>class: GenomicRatioSet 
dim: 862448 20 
metadata(0):
assays(2): M CN
rownames(862448): cg14817997 cg26928153 ... cg07587934 cg16855331
rowData names(0):
colnames(20): 202905570075_R02C01 202900540047_R01C01 ...
  204074230020_R05C01 204074230020_R06C01
colData names(9): Sample_ID Sample_Group ... yMed predictedSex
Annotation
  array: IlluminaHumanMethylationEPIC
  annotation: ilm10b4.hg19
Preprocessing
  Method: Raw (no normalization or bg correction)
  minfi version: 1.36.0
  Manifest version: 0.3.0</code></pre>
<p>Calculate M and beta values for downstream use in analysis and visulalisation.</p>
<pre class="r"><code># calculate M-values and beta values for downstream analysis and visualisation
mVals &lt;- getM(mSetSqFlt)
head(mVals[,1:5]) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">202905570075_R02C01</th>
<th align="right">202900540047_R01C01</th>
<th align="right">202900540047_R02C01</th>
<th align="right">202900540047_R03C01</th>
<th align="right">202900540047_R04C01</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cg14817997</td>
<td align="right">2.4883113</td>
<td align="right">2.225993</td>
<td align="right">1.7929596</td>
<td align="right">1.9652944</td>
<td align="right">2.0662500</td>
</tr>
<tr class="even">
<td align="left">cg26928153</td>
<td align="right">2.6937237</td>
<td align="right">2.279881</td>
<td align="right">1.9807748</td>
<td align="right">2.3369908</td>
<td align="right">2.3210052</td>
</tr>
<tr class="odd">
<td align="left">cg16269199</td>
<td align="right">1.7402849</td>
<td align="right">1.016383</td>
<td align="right">1.0947408</td>
<td align="right">1.0124683</td>
<td align="right">1.1203640</td>
</tr>
<tr class="even">
<td align="left">cg13869341</td>
<td align="right">2.8729742</td>
<td align="right">2.898040</td>
<td align="right">2.5100382</td>
<td align="right">2.5347448</td>
<td align="right">2.9068319</td>
</tr>
<tr class="odd">
<td align="left">cg14008030</td>
<td align="right">1.5839008</td>
<td align="right">1.271259</td>
<td align="right">0.9548359</td>
<td align="right">1.0340442</td>
<td align="right">0.1394236</td>
</tr>
<tr class="even">
<td align="left">cg12045430</td>
<td align="right">-0.7965316</td>
<td align="right">-0.760002</td>
<td align="right">-0.8327795</td>
<td align="right">-0.9423224</td>
<td align="right">-0.8896792</td>
</tr>
</tbody>
</table>
<pre class="r"><code>bVals &lt;- getBeta(mSetSqFlt)
head(bVals[,1:5]) %&gt;% knitr::kable()</code></pre>
<table style="width:100%;">
<colgroup>
<col width="9%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
<col width="18%" />
</colgroup>
<thead>
<tr class="header">
<th align="left"></th>
<th align="right">202905570075_R02C01</th>
<th align="right">202900540047_R01C01</th>
<th align="right">202900540047_R02C01</th>
<th align="right">202900540047_R03C01</th>
<th align="right">202900540047_R04C01</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">cg14817997</td>
<td align="right">0.8487417</td>
<td align="right">0.8238919</td>
<td align="right">0.7760484</td>
<td align="right">0.7961232</td>
<td align="right">0.8072463</td>
</tr>
<tr class="even">
<td align="left">cg26928153</td>
<td align="right">0.8661278</td>
<td align="right">0.8292460</td>
<td align="right">0.7978593</td>
<td align="right">0.8347784</td>
<td align="right">0.8332445</td>
</tr>
<tr class="odd">
<td align="left">cg16269199</td>
<td align="right">0.7696389</td>
<td align="right">0.6691854</td>
<td align="right">0.6810967</td>
<td align="right">0.6685844</td>
<td align="right">0.6849419</td>
</tr>
<tr class="even">
<td align="left">cg13869341</td>
<td align="right">0.8798905</td>
<td align="right">0.8817146</td>
<td align="right">0.8506650</td>
<td align="right">0.8528274</td>
<td align="right">0.8823487</td>
</tr>
<tr class="odd">
<td align="left">cg14008030</td>
<td align="right">0.7498620</td>
<td align="right">0.7070645</td>
<td align="right">0.6596740</td>
<td align="right">0.6718898</td>
<td align="right">0.5241415</td>
</tr>
<tr class="even">
<td align="left">cg12045430</td>
<td align="right">0.3653742</td>
<td align="right">0.3712651</td>
<td align="right">0.3595682</td>
<td align="right">0.3422760</td>
<td align="right">0.3505372</td>
</tr>
</tbody>
</table>
<p>Remove SNP probes and multi-mapping probes.</p>
<pre class="r"><code>mVals &lt;- DMRcate::rmSNPandCH(mVals, mafcut = 0, rmcrosshyb = TRUE, rmXY = FALSE)
bVals &lt;- DMRcate::rmSNPandCH(bVals, mafcut = 0, rmcrosshyb = TRUE, rmXY = FALSE)
dim(mVals)</code></pre>
<pre><code>[1] 749821     20</code></pre>
<p>Remove sex chromosome probes.</p>
<pre class="r"><code>mValsNoXY &lt;- DMRcate::rmSNPandCH(mVals, mafcut = 0, rmcrosshyb = TRUE, 
                                 rmXY = TRUE)
bValsNoXY &lt;- DMRcate::rmSNPandCH(bVals, mafcut = 0, rmcrosshyb = TRUE, 
                                 rmXY = TRUE)
dim(mValsNoXY)</code></pre>
<pre><code>[1] 732778     20</code></pre>
</div>
<div id="data-exploration-after-probe-filtering" class="section level2">
<h2>Data exploration (after probe filtering)</h2>
<p>After filtering, sorted cell types still cluster tightly together. The mixed samples are centered between the sorted cell types, which is indicative of the mixtures.</p>
<pre class="r"><code>mDat &lt;- mValsNoXY
mds &lt;- plotMDS(mDat, top = 1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group, 
              run = targets$Sample_run,
              source = targets$Sample_source,
              sex = targets$Sex)

p1 &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
       x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;) +
  ggtitle(&quot;All data&quot;) +
  scale_color_manual(values = pal)

p1</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-17-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>After filtering, the sorted cell types cluster tightly together and there is no evidence of a significant batch effect.</p>
<pre class="r"><code>cells &lt;- !(targets$Sample_Group %in% c(&quot;Raw&quot;))
mds &lt;- plotMDS(mDat[, cells], top = 1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group[cells], 
              run = targets$Sample_run[cells],
              source = targets$Sample_source[cells])

p1 &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
       x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;) +
  ggtitle(&quot;Cell types&quot;) +
  scale_color_manual(values = pal[-2])

p1</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-18-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-18-1">
Past versions of unnamed-chunk-18-1.png
</button>
</p>
<div id="fig-unnamed-chunk-18-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-18-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Calculate the percentage of variance explained by principal components.</p>
<pre class="r"><code>PCs &lt;- prcomp(t(mValsNoXY), scale = TRUE)
summary(PCs)</code></pre>
<pre><code>Importance of components:
                            PC1      PC2      PC3       PC4       PC5       PC6
Standard deviation     461.2224 385.6336 281.0357 225.07463 159.04781 145.86124
Proportion of Variance   0.2903   0.2029   0.1078   0.06913   0.03452   0.02903
Cumulative Proportion    0.2903   0.4933   0.6010   0.67016   0.70468   0.73372
                             PC7       PC8       PC9      PC10      PC11
Standard deviation     143.04165 138.20916 132.14354 127.12944 124.94789
Proportion of Variance   0.02792   0.02607   0.02383   0.02206   0.02131
Cumulative Proportion    0.76164   0.78771   0.81154   0.83359   0.85490
                            PC12      PC13      PC14      PC15      PC16
Standard deviation     121.33963 119.96664 116.90175 116.15774 114.55490
Proportion of Variance   0.02009   0.01964   0.01865   0.01841   0.01791
Cumulative Proportion    0.87499   0.89463   0.91328   0.93169   0.94960
                            PC17      PC18      PC19      PC20
Standard deviation     113.94919 111.17276 107.64848 4.526e-12
Proportion of Variance   0.01772   0.01687   0.01581 0.000e+00
Cumulative Proportion    0.96732   0.98419   1.00000 1.000e+00</code></pre>
<pre class="r"><code>dims &lt;- list(c(1,2), c(1,3), c(2,3))
pcvar &lt;- summary(PCs)$importance
p &lt;- vector(&quot;list&quot;, length(dims))

for(i in 1:length(p)){
  mds &lt;- plotMDS(mValsNoXY[, cells], top = 1000, gene.selection = &quot;common&quot;, 
                 plot = FALSE, dim.plot = dims[[i]])
  dat &lt;- tibble(x = mds$x, 
                y = mds$y, 
                sample = targets$Sample_Group[cells], 
                run = targets$Sample_run[cells],
                source = targets$Sample_source[cells])
  
  p[[i]] &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
    geom_point(aes(shape = run), size = 3) +
    labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
         x = glue(&quot;Principal component {dims[[i]][1]} ({round(pcvar[2,dims[[i]][1]]*100,1)}%)&quot;), 
         y = glue(&quot;Principal component {dims[[i]][2]} ({round(pcvar[2,dims[[i]][2]]*100,1)}%)&quot;)) +
    scale_color_manual(values = pal[-2]) + 
    theme(legend.direction = &quot;vertical&quot;)
  
}

(p[[1]] | p[[2]] | p[[3]]) + plot_layout(guides = &quot;collect&quot;) &amp;
  theme(legend.position = &quot;bottom&quot;)</code></pre>
<p><img src="figure/dataPreprocess.Rmd/unnamed-chunk-20-1.png" width="864" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-20-1">
Past versions of unnamed-chunk-20-1.png
</button>
</p>
<div id="fig-unnamed-chunk-20-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/dataPreprocess.Rmd/unnamed-chunk-20-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="save-processed-data" class="section level1">
<h1>Save processed data</h1>
<p>The data is of good quality and shows no evidence of unexpected sources of variation. Save the various data objects for faster downstream analysis.</p>
<pre class="r"><code>outFile &lt;- here(&quot;data/processedData.RData&quot;)
if(!file.exists(outFile)){
  save(annEPIC, mSetSqFlt, rgSet, mVals, bVals, targets, mValsNoXY, bValsNoXY, 
       pal, cells, file = outFile)
}</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] DMRcatedata_2.8.2                                  
 [2] ExperimentHub_1.16.0                               
 [3] AnnotationHub_2.22.0                               
 [4] BiocFileCache_1.14.0                               
 [5] dbplyr_2.1.0                                       
 [6] glue_1.4.2                                         
 [7] patchwork_1.1.1                                    
 [8] forcats_0.5.1                                      
 [9] dplyr_1.0.4                                        
[10] purrr_0.3.4                                        
[11] readr_1.4.0                                        
[12] tidyr_1.1.2                                        
[13] tibble_3.1.2                                       
[14] ggplot2_3.3.5                                      
[15] tidyverse_1.3.0                                    
[16] IlluminaHumanMethylationEPICmanifest_0.3.0         
[17] stringr_1.4.0                                      
[18] minfiData_0.36.0                                   
[19] IlluminaHumanMethylation450kmanifest_0.4.0         
[20] missMethyl_1.24.0                                  
[21] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
[22] IlluminaHumanMethylation450kanno.ilmn12.hg19_0.6.0 
[23] minfi_1.36.0                                       
[24] bumphunter_1.32.0                                  
[25] locfit_1.5-9.4                                     
[26] iterators_1.0.13                                   
[27] foreach_1.5.1                                      
[28] Biostrings_2.58.0                                  
[29] XVector_0.30.0                                     
[30] SummarizedExperiment_1.20.0                        
[31] Biobase_2.50.0                                     
[32] MatrixGenerics_1.2.1                               
[33] matrixStats_0.59.0                                 
[34] GenomicRanges_1.42.0                               
[35] GenomeInfoDb_1.26.7                                
[36] IRanges_2.24.1                                     
[37] S4Vectors_0.28.1                                   
[38] BiocGenerics_0.36.1                                
[39] limma_3.46.0                                       
[40] here_1.0.1                                         
[41] workflowr_1.6.2                                    

loaded via a namespace (and not attached):
  [1] utf8_1.2.1                    R.utils_2.10.1               
  [3] tidyselect_1.1.0              htmlwidgets_1.5.3            
  [5] RSQLite_2.2.5                 AnnotationDbi_1.52.0         
  [7] grid_4.0.2                    BiocParallel_1.24.1          
  [9] munsell_0.5.0                 codetools_0.2-18             
 [11] preprocessCore_1.52.1         statmod_1.4.35               
 [13] withr_2.4.2                   colorspace_2.0-2             
 [15] highr_0.8                     knitr_1.31                   
 [17] rstudioapi_0.13               labeling_0.4.2               
 [19] git2r_0.28.0                  GenomeInfoDbData_1.2.4       
 [21] bit64_4.0.5                   farver_2.1.0                 
 [23] rhdf5_2.34.0                  rprojroot_2.0.2              
 [25] vctrs_0.3.8                   generics_0.1.0               
 [27] xfun_0.23                     biovizBase_1.38.0            
 [29] R6_2.5.0                      illuminaio_0.32.0            
 [31] AnnotationFilter_1.14.0       bitops_1.0-7                 
 [33] rhdf5filters_1.2.0            cachem_1.0.4                 
 [35] reshape_0.8.8                 DelayedArray_0.16.3          
 [37] assertthat_0.2.1              bsseq_1.26.0                 
 [39] promises_1.2.0.1              scales_1.1.1                 
 [41] nnet_7.3-15                   gtable_0.3.0                 
 [43] ensembldb_2.14.0              rlang_0.4.11                 
 [45] genefilter_1.72.1             splines_4.0.2                
 [47] lazyeval_0.2.2                rtracklayer_1.50.0           
 [49] DSS_2.38.0                    GEOquery_2.58.0              
 [51] dichromat_2.0-0               checkmate_2.0.0              
 [53] broom_0.7.4                   BiocManager_1.30.10          
 [55] yaml_2.2.1                    reshape2_1.4.4               
 [57] modelr_0.1.8                  GenomicFeatures_1.42.1       
 [59] backports_1.2.1               httpuv_1.5.5                 
 [61] Hmisc_4.4-2                   tools_4.0.2                  
 [63] nor1mix_1.3-0                 ellipsis_0.3.2               
 [65] RColorBrewer_1.1-2            siggenes_1.64.0              
 [67] Rcpp_1.0.6                    plyr_1.8.6                   
 [69] base64enc_0.1-3               sparseMatrixStats_1.2.0      
 [71] progress_1.2.2                zlibbioc_1.36.0              
 [73] RCurl_1.98-1.3                prettyunits_1.1.1            
 [75] rpart_4.1-15                  openssl_1.4.3                
 [77] cluster_2.1.0                 haven_2.3.1                  
 [79] fs_1.5.0                      magrittr_2.0.1               
 [81] data.table_1.13.6             reprex_1.0.0                 
 [83] whisker_0.4                   ProtGenerics_1.22.0          
 [85] mime_0.10                     hms_1.0.0                    
 [87] evaluate_0.14                 xtable_1.8-4                 
 [89] XML_3.99-0.5                  jpeg_0.1-8.1                 
 [91] mclust_5.4.7                  readxl_1.3.1                 
 [93] gridExtra_2.3                 compiler_4.0.2               
 [95] biomaRt_2.46.3                crayon_1.4.1                 
 [97] R.oo_1.24.0                   htmltools_0.5.1.1            
 [99] later_1.1.0.1                 Formula_1.2-4                
[101] lubridate_1.7.9.2             DBI_1.1.1                    
[103] MASS_7.3-53.1                 DMRcate_2.4.1                
[105] rappdirs_0.3.3                Matrix_1.3-2                 
[107] permute_0.9-5                 cli_3.0.0                    
[109] R.methodsS3_1.8.1             quadprog_1.5-8               
[111] Gviz_1.34.1                   pkgconfig_2.0.3              
[113] GenomicAlignments_1.26.0      foreign_0.8-81               
[115] xml2_1.3.2                    annotate_1.68.0              
[117] rngtools_1.5                  multtest_2.46.0              
[119] beanplot_1.2                  rvest_0.3.6                  
[121] doRNG_1.8.2                   scrime_1.3.5                 
[123] VariantAnnotation_1.36.0      digest_0.6.27                
[125] rmarkdown_2.6                 base64_2.0                   
[127] cellranger_1.1.0              htmlTable_2.1.0              
[129] edgeR_3.32.1                  DelayedMatrixStats_1.12.3    
[131] curl_4.3                      shiny_1.6.0                  
[133] gtools_3.8.2                  Rsamtools_2.6.0              
[135] lifecycle_1.0.0               nlme_3.1-152                 
[137] jsonlite_1.7.2                Rhdf5lib_1.12.1              
[139] askpass_1.1                   BSgenome_1.58.0              
[141] fansi_0.5.0                   pillar_1.6.1                 
[143] lattice_0.20-41               fastmap_1.1.0                
[145] httr_1.4.2                    survival_3.2-7               
[147] interactiveDisplayBase_1.28.0 png_0.1-7                    
[149] BiocVersion_3.12.0            bit_4.0.4                    
[151] stringi_1.5.3                 HDF5Array_1.18.1             
[153] blob_1.2.1                    org.Hs.eg.db_3.12.0          
[155] latticeExtra_0.6-29           memoise_2.0.0.9000           </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
