<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Jovana Maksimovic" />

<meta name="date" content="2021-09-28" />

<title>Methylation profiling of paediatric Bronchoalveolar Lavage (BAL)</title>

<script src="site_libs/header-attrs-2.6/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>





<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">paed-BAL-meth-ref</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Methylation profiling of paediatric Bronchoalveolar Lavage (BAL)</h1>
<h3 class="subtitle">Estimating cell type proportions of paediatric BAL</h3>
<h4 class="author">Jovana Maksimovic</h4>
<h4 class="date">2021-09-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-09-28
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>paed-BAL-meth-ref/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210927code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210927)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210927code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210927)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedbalmethreftree2bd43cc6df0685982254d3eb180802a33ee74b19targetblank2bd43cca"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/tree/2bd43cc6df0685982254d3eb180802a33ee74b19" target="_blank">2bd43cc</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsatlassianpetermacorgaubitbucketscmospaedbalmethreftree2bd43cc6df0685982254d3eb180802a33ee74b19targetblank2bd43cca" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/tree/2bd43cc6df0685982254d3eb180802a33ee74b19" target="_blank">2bd43cc</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    code/deident.R
    Ignored:    data/idat/

Unstaged changes:
    Modified:   .gitignore

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/estimateCellProportions.Rmd</code>) and HTML (<code>docs/estimateCellProportions.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/2bd43cc6df0685982254d3eb180802a33ee74b19/analysis/estimateCellProportions.Rmd" target="_blank">2bd43cc</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-28
</td>
<td>
wflow_publish(“analysis/estimateCellProportions.Rmd”)
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/estimateCellProportions.html" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/1114da3d98ebc1f90be57f2626e71b16d7ea20f7/analysis/estimateCellProportions.Rmd" target="_blank">1114da3</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
<td>
wflow_publish(c(“analysis/index.Rmd”, “analysis/dataPreprocess.Rmd”,
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="data-input" class="section level1">
<h1>Data input</h1>
<p>Load all necessary analysis packages.</p>
<pre class="r"><code>library(here)
library(workflowr)
library(glue)
#Load Packages Required for Analysis
library(limma)
library(minfi)
library(matrixStats)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(IlluminaHumanMethylationEPICmanifest)
library(FlowSorted.Blood.EPIC)
library(ggplot2)
library(ExperimentHub)
library(reshape2)
library(tidyverse)
library(patchwork)</code></pre>
<p>Load raw and processed data objects generated by exploratory analysis.</p>
<pre class="r"><code># load data objects
load(here(&quot;data/processedData.RData&quot;))
# load modified cell type estimation function
source(here(&quot;code/functions.R&quot;))</code></pre>
<p>As expected, we see clear clustering by cell types.</p>
<pre class="r"><code>mds &lt;- plotMDS(mVals[, cells], top = 1000, gene.selection=&quot;common&quot;, plot = FALSE)
dat &lt;- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group[cells], 
              run = targets$Sample_run[cells],
              ID = targets$Sample_ID[cells])

p &lt;- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = &quot;Sample type&quot;, shape = &quot;Run&quot;,
       x = &quot;Principal component 1&quot;, 
       y = &quot;Principal component 2&quot;) +
  ggtitle(&quot;Cell types&quot;) +
  scale_color_manual(values = pal)
p</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-3-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Identify cell type discriminating probes using the <em>any</em> and <em>both</em> method (<em>both</em> selects an equal number (50) of probes (with F-stat p-value &lt; 1E-8) with the greatest magnitude of effect from the hyper and hypo methylated sides; or <em>any</em>, which selects the 100 probes (with F-stat p-value &lt; 1E-8) with the greatest magnitude of difference regardless of direction of effect.).</p>
<pre class="r"><code>mSetSqFlt$CellType &lt;- as.character(targets$Sample_Group)

pAny &lt;- minfi:::pickCompProbes(mSet = mSetSqFlt[,cells], probeSelect = &quot;any&quot;,
                       numProbes = 50, compositeCellType = &quot;Lavage&quot;, 
                       cellTypes = unique(targets$Sample_Group[cells]))
pBoth &lt;- minfi:::pickCompProbes(mSet = mSetSqFlt[,cells], probeSelect = &quot;both&quot;,
                       numProbes = 50, compositeCellType = &quot;Lavage&quot;, 
                       cellTypes = unique(targets$Sample_Group[cells]))</code></pre>
<p>Use a heatmap to visulalise the methylation of the cell type discriminating probes in the cell sorted samples.</p>
<pre class="r"><code>par(mfrow=c(1,2))
NMF::aheatmap(bVals[rownames(bVals) %in% rownames(pAny$coefEsts), cells],
         annCol = list(CellType = as.character(targets$Sample_Group[cells])), 
         labCol = NA, labRow = NA, annColors = list(pal[-2]),
         main=&quot;A. ProbeSelect = Any&quot;)

NMF::aheatmap(bVals[rownames(bVals) %in% rownames(pBoth$coefEsts), cells],
         annCol = list(CellType = as.character(targets$Sample_Group[cells])), 
         labCol = NA, labRow = NA, annColors = list(pal[-2]),
         main=&quot;B. ProbeSelect = Both&quot;)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-5-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-5-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="estimate-cell-type-proportions-of-bal-samples" class="section level1">
<h1>Estimate cell type proportions of BAL samples</h1>
<div id="artificial-mixture" class="section level2">
<h2>Artificial mixture</h2>
<p>We will use blood immune cell mixtures with known cell type proportions published by Salas et al. 2018 to test the accuracy of cell type proportion estimates derived using our reference library.</p>
<pre class="r"><code>hub &lt;- ExperimentHub()  </code></pre>
<pre><code>snapshotDate(): 2020-10-27</code></pre>
<pre class="r"><code>query(hub, &quot;FlowSorted.Blood.EPIC&quot;)  </code></pre>
<pre><code>ExperimentHub with 1 record
# snapshotDate(): 2020-10-27
# names(): EH1136
# package(): FlowSorted.Blood.EPIC
# $dataprovider: GEO
# $species: Homo sapiens
# $rdataclass: RGChannelSet
# $rdatadateadded: 2018-04-20
# $title: FlowSorted.Blood.EPIC: Illumina Human Methylation data from EPIC o...
# $description: The FlowSorted.Blood.EPIC package contains Illumina HumanMet...
# $taxonomyid: 9606
# $genome: hg19
# $sourcetype: tar.gz
# $sourceurl: https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE110554
# $sourcesize: NA
# $tags: c(&quot;ExperimentData&quot;, &quot;Homo_sapiens_Data&quot;, &quot;Tissue&quot;,
#   &quot;MicroarrayData&quot;, &quot;Genome&quot;, &quot;TissueMicroarrayData&quot;,
#   &quot;MethylationArrayData&quot;) 
# retrieve record with &#39;object[[&quot;EH1136&quot;]]&#39; </code></pre>
<pre class="r"><code>FlowSorted.Blood.EPIC &lt;- hub[[&quot;EH1136&quot;]]</code></pre>
<pre><code>see ?FlowSorted.Blood.EPIC and browseVignettes(&#39;FlowSorted.Blood.EPIC&#39;) for documentation</code></pre>
<pre><code>loading from cache</code></pre>
<pre class="r"><code># separate the reference from the testing dataset  
RGsetTargets &lt;- FlowSorted.Blood.EPIC[,FlowSorted.Blood.EPIC$CellType == &quot;MIX&quot;]
mixReal &lt;- as.matrix(colData(RGsetTargets)[,12:17])/100</code></pre>
<pre class="r"><code>mixDat &lt;- melt(mixReal)
colnames(mixDat) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p &lt;- ggplot(mixDat, aes(sample)) + 
  geom_bar(aes(fill = cell, weight=proportion)) +
  scale_x_discrete(breaks = waiver(), labels=1:nrow(mixReal)) +
  ggtitle(&quot;Artificially created cell mixtures&quot;) +
  labs(y = &quot;Proportion&quot;, x = &quot;Mixture&quot;, fill = &quot;Cell type&quot;)
p</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-7-1">
Past versions of unnamed-chunk-7-1.png
</button>
</p>
<div id="fig-unnamed-chunk-7-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-7-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We have not sorted our BAL lymphocytes into B cells, T cells and NK cells, so the “true” lymphocyte value we will compare to is the sum of the proportions of those cells in the artificial mixture.</p>
<pre class="r"><code>mixSum &lt;- data.frame(mixReal[,5:6], Lymph = rowSums(mixReal[,1:4]))
mixSumDat &lt;- melt(as.matrix(mixSum))
colnames(mixSumDat) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p &lt;- ggplot(mixSumDat, aes(sample)) + 
  geom_bar(aes(fill = cell, weight = proportion)) +
    scale_x_discrete(breaks = waiver(), labels = 1:nrow(mixReal)) +
    ggtitle(&quot;Artificially created cell mixtures&quot;) +
  labs(y = &quot;Proportion&quot;, x = &quot;Mixture&quot;, fill = &quot;Cell type&quot;)
p</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-8-1">
Past versions of unnamed-chunk-8-1.png
</button>
</p>
<div id="fig-unnamed-chunk-8-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-8-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Estimate cell type proportions in artificial mixture using <em>both</em> option.</p>
<pre class="r"><code>lavageRef &lt;- rgSet[,cells]
colData(lavageRef)$CellType &lt;- colData(lavageRef)$Sample_Group
mixEstBoth &lt;- estimateCellCounts2Mod(rgSet = RGsetTargets,
                                  compositeCellType = &quot;Lavage&quot;,
                                  processMethod = &quot;preprocessQuantile&quot;,
                                  probeSelect = &quot;both&quot;,
                                  cellTypes = unique(targets$Sample_Group[cells]),
                                  referencePlatform =
                                    &quot;IlluminaHumanMethylationEPIC&quot;,
                                  referenceset = &quot;lavageRef&quot;,
                                  IDOLOptimizedCpGs = NULL,
                                  returnAll = FALSE,
                                  meanPlot = FALSE,
                                  keepProbes = rownames(mValsNoXY))</code></pre>
<pre><code>[estimateCellCounts2] Combining user data with reference (flow sorted) data.</code></pre>
<pre><code>Warning in DataFrame(sampleNames = c(colnames(rgSet),
colnames(referenceRGset)), : &#39;stringsAsFactors&#39; is ignored</code></pre>
<pre><code>[estimateCellCounts2] Processing user and reference data together.</code></pre>
<pre><code>[preprocessQuantile] Mapping to genome.</code></pre>
<pre><code>Warning in .getSex(CN = CN, xIndex = xIndex, yIndex = yIndex, cutoff = cutoff):
An inconsistency was encountered while determining sex. One possibility is
that only one sex is present. We recommend further checks, for example with the
plotSex function.</code></pre>
<pre><code>[preprocessQuantile] Fixing outliers.</code></pre>
<pre><code>[preprocessQuantile] Quantile normalizing.</code></pre>
<pre><code>[estimateCellCounts2] Picking probes for composition estimation.</code></pre>
<pre><code>[estimateCellCounts2] Estimating composition.</code></pre>
<pre class="r"><code>mixEstBoth$counts</code></pre>
<pre><code>                    EpithelialCell    Macrophage Granulocyte Lymphocyte
201868590193_R01C01   0.000000e+00  9.126279e-02   0.3616930  0.5660512
201868590243_R02C01   0.000000e+00 -4.269868e-20   0.7388455  0.2612947
201868590267_R01C01   0.000000e+00  1.915281e-02   0.7804746  0.2008515
201868590267_R05C01   0.000000e+00 -3.189628e-20   0.6796117  0.3189449
201869680008_R01C01   2.404706e-04  8.807077e-02   0.2596974  0.6657974
201869680008_R03C01  -1.179253e-18  0.000000e+00   0.7136450  0.2817510
201869680008_R06C01   0.000000e+00  7.828989e-02   0.3636083  0.5640165
201869680030_R03C01   0.000000e+00 -4.890270e-20   0.7014649  0.3042240
201869680030_R07C01   0.000000e+00 -5.957902e-20   0.7772578  0.2200175
201870610056_R01C01   0.000000e+00  3.605620e-02   0.2957907  0.6859857
201870610056_R03C01   0.000000e+00  1.131711e-01   0.2196631  0.6858875
201870610111_R03C01   0.000000e+00  1.024043e-01   0.4025289  0.5060127</code></pre>
<pre class="r"><code>mixBoth &lt;- reshape2::melt(mixEstBoth$counts)
colnames(mixBoth) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p1 &lt;- ggplot(mixBoth, aes(x = sample, fill = cell)) + 
  geom_bar(aes(weight = proportion)) +
  ggtitle(&quot;Cell type proportion estimates&quot;, 
          subtitle = &quot;Probe selection: Both&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Proportion&quot;, fill = &quot;Cell type&quot;) + 
  scale_x_discrete(breaks = waiver(), labels = 1:nrow(mixReal)) +
  scale_fill_manual(values = pal)
p1</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-10-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-10-1">
Past versions of unnamed-chunk-10-1.png
</button>
</p>
<div id="fig-unnamed-chunk-10-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-10-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>(p | p1) + 
  plot_layout(guides = &quot;collect&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;,
        legend.box = &quot;vertical&quot;)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-11-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Estimate cell type proportions in artificial mixture using <em>any</em> option.</p>
<pre class="r"><code>mixEstAny &lt;- estimateCellCounts2Mod(rgSet = RGsetTargets, 
                                 compositeCellType = &quot;Lavage&quot;,   
                                processMethod = &quot;preprocessQuantile&quot;,  
                                probeSelect = &quot;any&quot;,  
                                cellTypes = unique(targets$Sample_Group[cells]),  
                                referencePlatform =   
                                &quot;IlluminaHumanMethylationEPIC&quot;,  
                                referenceset = &quot;lavageRef&quot;,  
                                IDOLOptimizedCpGs = NULL,   
                                returnAll = FALSE,
                                meanPlot = FALSE,
                                keepProbes = rownames(mValsNoXY))</code></pre>
<pre><code>[estimateCellCounts2] Combining user data with reference (flow sorted) data.</code></pre>
<pre><code>Warning in DataFrame(sampleNames = c(colnames(rgSet),
colnames(referenceRGset)), : &#39;stringsAsFactors&#39; is ignored</code></pre>
<pre><code>[estimateCellCounts2] Processing user and reference data together.</code></pre>
<pre><code>[preprocessQuantile] Mapping to genome.</code></pre>
<pre><code>Warning in .getSex(CN = CN, xIndex = xIndex, yIndex = yIndex, cutoff = cutoff):
An inconsistency was encountered while determining sex. One possibility is
that only one sex is present. We recommend further checks, for example with the
plotSex function.</code></pre>
<pre><code>[preprocessQuantile] Fixing outliers.</code></pre>
<pre><code>[preprocessQuantile] Quantile normalizing.</code></pre>
<pre><code>[estimateCellCounts2] Picking probes for composition estimation.</code></pre>
<pre><code>[estimateCellCounts2] Estimating composition.</code></pre>
<pre class="r"><code>mixEstAny$counts</code></pre>
<pre><code>                    EpithelialCell  Macrophage Granulocyte Lymphocyte
201868590193_R01C01   0.000000e+00 0.147918242   0.3053763  0.5746209
201868590243_R02C01   0.000000e+00 0.006390512   0.7234165  0.2676621
201868590267_R01C01   1.734723e-18 0.048680044   0.7572886  0.2027038
201868590267_R05C01   0.000000e+00 0.019133044   0.6613300  0.3209722
201869680008_R01C01   0.000000e+00 0.162845311   0.2165590  0.6574659
201869680008_R03C01   0.000000e+00 0.012635790   0.7008658  0.2832374
201869680008_R06C01   0.000000e+00 0.135170214   0.3300484  0.5587240
201869680030_R03C01  -1.734723e-18 0.004996861   0.6784126  0.3156032
201869680030_R07C01   0.000000e+00 0.004925952   0.7677886  0.2238971
201870610056_R01C01   0.000000e+00 0.079967402   0.2388535  0.6965378
201870610056_R03C01   0.000000e+00 0.184260959   0.1537695  0.6933920
201870610111_R03C01   0.000000e+00 0.166083551   0.3528550  0.5060989</code></pre>
<pre class="r"><code>mixAny &lt;- reshape2::melt(mixEstAny$counts)
colnames(mixAny) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p2 &lt;- ggplot(mixAny, aes(x = sample, fill = cell)) + 
  geom_bar(aes(weight = proportion)) +
  ggtitle(&quot;Cell type proportion estimates&quot;, 
          subtitle = &quot;Probe selection: Any&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Proportion&quot;, fill = &quot;Cell type&quot;) + 
  scale_x_discrete(breaks = waiver(), labels = 1:nrow(mixReal)) +
  scale_fill_manual(values = pal)
p2</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-13-1">
Past versions of unnamed-chunk-13-1.png
</button>
</p>
<div id="fig-unnamed-chunk-13-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-13-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>(p | p2) + 
  plot_layout(guides = &quot;collect&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;,
        legend.box = &quot;vertical&quot;)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-14-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>mixSumDat$est &lt;- &quot;truth&quot;
mixSumDat$num &lt;- rownames(mixSumDat)
mixBoth$est &lt;- &quot;both&quot;
mixBoth$num &lt;- rownames(mixBoth)
mixAny$est &lt;- &quot;any&quot;
mixAny$num &lt;- rownames(mixAny)

fullDat &lt;- rbind(mixSumDat, mixBoth, mixAny)
fullDat$cell &lt;- fct_recode(fullDat$cell, 
                           Lymphocyte = &quot;Lymph&quot;, 
                           Granulocyte = &quot;Neu&quot;, 
                           Macrophage = &quot;Mono&quot;)
fullDat &lt;- fullDat[fullDat$cell != &quot;EpithelialCell&quot;,]
fullDat$cell &lt;- fct_drop(fullDat$cell)

p1 &lt;- ggplot(fullDat, aes(x = sample, y = proportion, fill = est)) + 
  geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
  facet_grid(.~ cell, scales = &quot;free_x&quot;) +
  labs(x = &quot;Mixture&quot;, y = &quot;Proportion&quot;, fill = &quot;Source&quot;) + 
  scale_x_discrete(breaks = waiver(), labels = 1:nrow(mixReal)) +
  theme(axis.text.x = element_text(size = 8))

p1</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-15-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-15-1">
Past versions of unnamed-chunk-15-1.png
</button>
</p>
<div id="fig-unnamed-chunk-15-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-15-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>We can see that the estimates of granulocyte and lymphocute proportions made using our BAL reference panel are highly correlated with the true proportions regardless of how the discrimination probe set is chosen. Although there is evidence of slight overestimation of granulocytes in some samples. For example, mixtures 1, 5, 7, 10, 11 and 12 are dominated by &gt;50% lymphocytes and in those samples the proportion of granulocytes is slightly overestimated. This discrepancy is likely due to the fact that the mixture samples contain only neutrophils whearas our refernce panel contains all granulocyte types. However, despite this shortcoming, the cell type proportion estimates are still very accurate. This suggests that our BAL reference panel, in combination with the Houseman algorithm, should be able to accurately estimate cell type proportions in our patient BAL samples.</p>
<pre class="r"><code>dat1 &lt;- tibble(x = rowMeans(cbind(mixSum[,&quot;Neu&quot;], 
                                 mixEstAny$counts[,&quot;Granulocyte&quot;])),
              y = (mixSum[,&quot;Neu&quot;] - mixEstAny$counts[,&quot;Granulocyte&quot;]))
c1 &lt;- cor.test(mixSum[,&#39;Neu&#39;], mixEstAny$counts[,&#39;Granulocyte&#39;])
p1 &lt;- ggplot(dat1, aes(x = x, y = y)) +
  geom_text(label = 1:nrow(mixSum), size = 3) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Truth - Estimate)&quot;) +
  coord_cartesian(ylim = c(-0.2, 0.2)) +
  ggtitle(&quot;Granulocytes&quot;, subtitle = &quot;Probe selection: Any&quot;) +
  annotate(&quot;text&quot;, x = 0.1, y = 0.15, hjust = 0, size = 3,
           label = glue::glue(&quot;Pearson correlation = {round(c1$estimate, 4)}
                               p-value = {signif(c1$p.value, 4)}&quot;))
  

dat2 &lt;- tibble(x = rowMeans(cbind(mixSum[,&quot;Lymph&quot;], 
                                 mixEstAny$counts[,&quot;Lymphocyte&quot;])),
              y = (mixSum[,&quot;Lymph&quot;] - mixEstAny$counts[,&quot;Lymphocyte&quot;]))
c2 &lt;- cor.test(mixSum[,&#39;Lymph&#39;], mixEstAny$counts[,&#39;Lymphocyte&#39;])
p2 &lt;- ggplot(dat2, aes(x = x, y = y)) +
  geom_text(label = 1:nrow(mixSum), size = 3) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Truth - Estimate)&quot;) +
  coord_cartesian(ylim = c(-0.2, 0.2)) +
  ggtitle(&quot;Lymphocytes&quot;, subtitle = &quot;Probe selection: Any&quot;) +
  annotate(&quot;text&quot;, x = 0.1, y = 0.15, hjust = 0, size = 3,
           label = glue::glue(&quot;Pearson correlation = {round(c2$estimate, 4)}
                               p-value = {signif(c2$p.value, 4)}&quot;))

dat3 &lt;- tibble(x = rowMeans(cbind(mixSum[,&quot;Neu&quot;], 
                                 mixEstBoth$counts[,&quot;Granulocyte&quot;])),
              y = (mixSum[,&quot;Neu&quot;] - mixEstBoth$counts[,&quot;Granulocyte&quot;]))
c3 &lt;- cor.test(mixSum[,&#39;Neu&#39;], mixEstBoth$counts[,&#39;Granulocyte&#39;])
p3 &lt;- ggplot(dat3, aes(x = x, y = y)) +
  geom_text(label = 1:nrow(mixSum), size = 3) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Truth - Estimate)&quot;) +
  coord_cartesian(ylim = c(-0.2, 0.2)) +
  ggtitle(&quot;Granulocytes&quot;, subtitle = &quot;Probe selection: Both&quot;) +
  annotate(&quot;text&quot;, x = 0.1, y = 0.15, hjust = 0, size = 3,
           label = glue::glue(&quot;Pearson correlation = {round(c3$estimate, 4)}
                               p-value = {signif(c3$p.value, 4)}&quot;))

dat4 &lt;- tibble(x = rowMeans(cbind(mixSum[,&quot;Lymph&quot;], 
                                 mixEstBoth$counts[,&quot;Lymphocyte&quot;])),
              y = (mixSum[,&quot;Lymph&quot;] - mixEstBoth$counts[,&quot;Lymphocyte&quot;]))
c4 &lt;- cor.test(mixSum[,&#39;Lymph&#39;], mixEstBoth$counts[,&#39;Lymphocyte&#39;])
p4 &lt;- ggplot(dat4, aes(x = x, y = y)) +
  geom_text(label = 1:nrow(mixSum), size = 3) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Truth - Estimate)&quot;) +
  coord_cartesian(ylim = c(-0.2, 0.2)) +
  ggtitle(&quot;Lymphocytes&quot;, subtitle = &quot;Probe selection: Both&quot;) +
  annotate(&quot;text&quot;, x = 0.1, y = 0.15, hjust = 0, size = 3,
           label = glue::glue(&quot;Pearson correlation = {round(c4$estimate, 4)}
                               p-value = {signif(c4$p.value, 4)}&quot;))

(p1 | p2) / (p3 | p4)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-16-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-16-1">
Past versions of unnamed-chunk-16-1.png
</button>
</p>
<div id="fig-unnamed-chunk-16-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-16-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>If we look at MDS plots of the mixture data, as well as of the known cell type proportions, we can see that the primary source of variation is the proportion of neutrophils vs. lymphoctes. This overlaps with the the samples where we are seeing slight over estimation of granulocyte proportions.</p>
<pre class="r"><code>mSetTargets &lt;- preprocessRaw(RGsetTargets)
mValsTargets &lt;- getM(mSetTargets)

mds &lt;- plotMDS(mValsTargets, top = 1000, gene.selection = &quot;common&quot;, 
               plot = FALSE)
dat &lt;- tibble(x = mds$x, y = mds$y, sample = 1:12)
p1 &lt;- ggplot(dat, aes(x = x, y = y)) +
  geom_text(aes(label = sample)) +
  labs(x = &quot;Principal Component 1&quot;, y = &quot;Principal Component 2&quot;)

mds &lt;- plotMDS(mixReal, top = 1000, gene.selection = &quot;common&quot;, 
               plot = FALSE)
dat &lt;- tibble(x = mds$x, y = mds$y, sample = colnames(mixReal))
p2 &lt;- ggplot(dat, aes(x = x, y = y)) +
  geom_text(aes(label = sample)) +
  labs(x = &quot;Principal Component 1&quot;, y = &quot;Principal Component 2&quot;)

(p1 | p2)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-17-1">
Past versions of unnamed-chunk-17-1.png
</button>
</p>
<div id="fig-unnamed-chunk-17-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-17-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="raw-bal-samples" class="section level2">
<h2>Raw BAL samples</h2>
<p>We can estimate the proportions of cell types in each of our raw BAL samples using our reference panel. To select the most informative cell type probes from our reference panel we can either set the ’<code>probeSelect</code> paremeter to <em>both</em>, which selects an equal number (50) of probes (with F-stat p-value &lt; 1E-8) with the greatest magnitude of effect from the hyper and hypo methylated sides; or <em>any</em>, which selects the 100 probes (with F-stat p-value &lt; 1E-8) with the greatest magnitude of difference regardless of direction of effect.</p>
<p>Get estimates for proportion of each cell type using the <em>any</em> option.</p>
<pre class="r"><code>patientSamps &lt;- rgSet[, targets$Sample_Group == &quot;Raw&quot;]
sampleNames(patientSamps) &lt;- targets$Sample_source[targets$Sample_Group == &quot;Raw&quot;]
cellEstAny &lt;- estimateCellCounts2Mod(rgSet = patientSamps, 
                                  compositeCellType = &quot;Lavage&quot;,   
                                  processMethod = &quot;preprocessQuantile&quot;,  
                                  probeSelect = &quot;any&quot;,  
                                  cellTypes = unique(targets$Sample_Group[cells]),  
                                  referencePlatform = 
                                    &quot;IlluminaHumanMethylationEPIC&quot;,  
                                  referenceset = &quot;lavageRef&quot;,  
                                  IDOLOptimizedCpGs = NULL,   
                                  returnAll = TRUE,
                                  meanPlot = FALSE,
                                  keepProbes = rownames(mValsNoXY))
cellEstAny$counts</code></pre>
<pre><code>     EpithelialCell Macrophage Granulocyte Lymphocyte
CF4      0.09793734 0.66758315   0.1464392 0.12014193
CF1      0.09032793 0.01217868   0.8898728 0.01503895
CF3      0.21500314 0.37264030   0.2236391 0.22976338
CF2      0.11447882 0.30504955   0.4358487 0.20458003
CF5      0.05676325 0.51112862   0.3566726 0.11446217
CTRL     0.12935134 0.14927553   0.5845740 0.18375917</code></pre>
<pre class="r"><code>estAny &lt;- reshape2::melt(cellEstAny$counts)
colnames(estAny) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p1 &lt;- ggplot(estAny, aes(x = sample, fill = cell)) + 
  geom_bar(aes(weight = proportion)) +
  ggtitle(&quot;Cell type proportion estimates&quot;, 
          subtitle = &quot;Probe selection: Any&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Proportion&quot;, fill = &quot;Cell type&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
        legend.position = &quot;bottom&quot;) +
  scale_fill_manual(values = pal)
p1</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-19-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-19-1">
Past versions of unnamed-chunk-19-1.png
</button>
</p>
<div id="fig-unnamed-chunk-19-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-19-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p>Get estimates for proportion of each cell type using the <em>both</em> option.</p>
<pre class="r"><code>cellEstBoth &lt;- estimateCellCounts2Mod(rgSet = patientSamps, 
                                   compositeCellType = &quot;Lavage&quot;,   
                                   processMethod = &quot;preprocessQuantile&quot;,  
                                   probeSelect = &quot;both&quot;,  
                                   cellTypes = unique(targets$Sample_Group[cells]),  
                                   referencePlatform =   
                                     &quot;IlluminaHumanMethylationEPIC&quot;,  
                                   referenceset = &quot;lavageRef&quot;,  
                                   IDOLOptimizedCpGs = NULL,   
                                   returnAll = TRUE,
                                   meanPlot = FALSE,
                                   keepProbes = rownames(mValsNoXY))
cellEstBoth$counts</code></pre>
<pre><code>     EpithelialCell  Macrophage Granulocyte Lymphocyte
CF4      0.10056117 0.687692615   0.1564780 0.09890711
CF1      0.08731582 0.006252298   0.8994798 0.01570180
CF3      0.20822410 0.393715752   0.2552786 0.20976207
CF2      0.11237742 0.338996251   0.4795570 0.16823990
CF5      0.05805766 0.526853480   0.3595676 0.09876533
CTRL     0.13196682 0.136015253   0.6171474 0.16751309</code></pre>
<pre class="r"><code>estBoth &lt;- reshape2::melt(cellEstBoth$counts)
colnames(estBoth) &lt;- c(&quot;sample&quot;,&quot;cell&quot;,&quot;proportion&quot;)

p2 &lt;- ggplot(estBoth, aes(x = sample, fill = cell)) + 
  geom_bar(aes(weight = proportion)) +
  ggtitle(&quot;Cell type proportion estimates&quot;, 
          subtitle = &quot;Probe selection: Both&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Proportion&quot;, fill = &quot;Cell type&quot;) +
  theme(axis.text.x = element_text(angle = 45, hjust = 1, size = 7),
        legend.position = &quot;bottom&quot;) +
  scale_fill_manual(values = pal)
p2</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-21-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-21-1">
Past versions of unnamed-chunk-21-1.png
</button>
</p>
<div id="fig-unnamed-chunk-21-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-21-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>fullDat &lt;- rbind(estBoth, estAny)
fullDat$est &lt;- rep(c(&quot;both&quot;,&quot;any&quot;), each = nrow(estAny))

p1 &lt;- ggplot(fullDat, aes(x = sample, y = proportion, fill = est)) + 
  geom_bar(stat = &quot;identity&quot;, position = &quot;dodge&quot;) +
  facet_grid(.~ cell, scales = &quot;free_x&quot;) +
  labs(x = &quot;Sample&quot;, y = &quot;Proportion&quot;, fill = &quot;Source&quot;) +
  theme(axis.text.x = element_text(size = 8, angle = 90, hjust = 1, vjust = 0.5))

p1</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-22-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-22-1">
Past versions of unnamed-chunk-22-1.png
</button>
</p>
<div id="fig-unnamed-chunk-22-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-22-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="compare-to-flow-cytometry-data" class="section level4">
<h4>Compare to flow-cytometry data</h4>
<pre class="r"><code>samps &lt;- targets$Sample_source[targets$Sample_Group == &quot;Raw&quot;]
melt(read.csv(here(&quot;data/Flow-Data-for-Reference-Panel-Scaled.csv&quot;),
              stringsAsFactors = FALSE)) %&gt;%
  mutate(variable = fct_recode(variable, CTRL = &quot;Control&quot;)) %&gt;%
  inner_join(melt(read.csv(here(&quot;data/Flow-Data-for-Reference-Panel-Original.csv&quot;),
                           stringsAsFactors = FALSE)) %&gt;%
  mutate(variable = fct_recode(variable, CTRL = &quot;Control&quot;)), 
             by = c(&quot;X&quot;, &quot;variable&quot;)) %&gt;%
  rename(cell = X, 
         sample = variable, 
         scaled = value.x, 
         original = value.y) %&gt;%
  mutate(scaled = scaled / 100,
         original = original / 100) %&gt;%
  inner_join(estBoth[estBoth$sample %in% samps,] %&gt;%
               rename(both = proportion) %&gt;%
               inner_join(estAny[estBoth$sample %in% samps,] %&gt;%
                            rename(any = proportion))  %&gt;% 
               mutate(cell = gsub(&quot;EpithelialCell&quot;,&quot;AEC&quot;, cell),
                      cell = gsub(&quot;Macrophage&quot;,&quot;macrophages&quot;, cell),
                      cell = gsub(&quot;Granulocyte&quot;,&quot;granulocytes&quot;, cell),
                      cell = gsub(&quot;Lymphocyte&quot;,&quot;lymphocytes&quot;, cell))) %&gt;%
  pivot_longer(cols = c(scaled, original)) -&gt; dat

p1 &lt;- ggplot(dat, aes(x = rowMeans(cbind(value, any)), 
                      y = value - any, colour = cell)) +
  geom_point(aes(shape = name)) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  facet_wrap(vars(sample), ncol = 3, nrow = 2) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Flow - Meth. est.)&quot;,
       colour = &quot;Cell Type&quot;, shape = &quot;Flow Data&quot;) +
  scale_shape_manual(values = c(1,4)) +
  ggtitle(&quot;Probe selection: Any&quot;)

p2 &lt;- ggplot(dat, aes(x = rowMeans(cbind(value, both)), 
                      y = value - both, colour = cell)) +
  geom_point(aes(shape = name)) +
  geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
  facet_wrap(vars(sample), ncol = 3, nrow = 2) +
  labs(x = &quot;Mean&quot;, y = &quot;Difference (Flow - Meth. est.)&quot;,
       colour = &quot;Cell Type&quot;, shape = &quot;Flow Data&quot;) +
  scale_shape_manual(values = c(1,4)) +
  ggtitle(&quot;Probe selection: Both&quot;)

p1 / p2 + plot_layout(guides = &quot;collect&quot;)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-23-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-23-1">
Past versions of unnamed-chunk-23-1.png
</button>
</p>
<div id="fig-unnamed-chunk-23-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-23-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>unique(dat$sample)</code></pre>
<pre><code>[1] CF1  CF2  CTRL CF3  CF4  CF5 
Levels: AF CF1 CF2 CTRL CF3 CF4 CF5</code></pre>
<pre class="r"><code>samps &lt;- as.character(unique(dat$sample))
p &lt;- vector(&quot;list&quot;, length(samps))

for(i in 1:length(samps)){
  dat1 &lt;- filter(dat, sample == samps[i])
  c1 &lt;- cor.test(dat1$any, dat1$value)
  p[[i]] &lt;- ggplot(dat1, aes(x = rowMeans(cbind(value, any)), 
                         y = value - any, colour = cell)) +
    geom_point(aes(shape = name)) +
    scale_shape_manual(values = c(1,4)) +
    geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
    labs(x = &quot;Mean&quot;, y = &quot;Difference (Flow - Meth. est.)&quot;,
         colour = &quot;Cell type&quot;) +
    coord_cartesian(ylim = c(-0.5, 0.5), xlim = c(0, 1)) +
    ggtitle({samps[i]}) +
    annotate(&quot;text&quot;, x = 0, y = 0.45, hjust = 0, size = 3,
             label = glue::glue(&quot;Pearson correlation = {round(c1$estimate, 4)}
                               p-value = {signif(c1$p.value, 4)}&quot;))
  
}

p1 &lt;- wrap_plots(p, ncol = 3, guides = &quot;collect&quot;) +
  plot_annotation(title = &quot;Probe selection: Any&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;, legend.box = &quot;vertical&quot;)

for(i in 1:length(samps)){
  dat1 &lt;- filter(dat, sample == samps[i])
  c1 &lt;- cor.test(dat1$both, dat1$value)
  p[[i]] &lt;- ggplot(dat1, aes(x = rowMeans(cbind(value, both)), 
                         y = value - both, colour = cell)) +
    geom_point(aes(shape = name)) +
    scale_shape_manual(values = c(1,4)) +
    geom_hline(yintercept = 0, linetype = &quot;dashed&quot;, colour = &quot;red&quot;) +
    labs(x = &quot;Mean&quot;, y = &quot;Difference (Flow - Meth. est.)&quot;,
         colour = &quot;Cell type&quot;) +
    coord_cartesian(ylim = c(-0.5, 0.5), xlim = c(0, 1)) +
    ggtitle(samps[i]) +
    annotate(&quot;text&quot;, x = 0, y = 0.45, hjust = 0, size = 3,
             label = glue::glue(&quot;Pearson correlation = {round(c1$estimate, 4)}
                               p-value = {signif(c1$p.value, 4)}&quot;))
  
}

p2 &lt;- wrap_plots(p, ncol = 3, guides = &quot;collect&quot;) +
  plot_annotation(title = &quot;Probe selection: Both&quot;) &amp; 
  theme(legend.position = &quot;bottom&quot;, legend.box = &quot;vertical&quot;)

wrap_elements(p1) / 
    wrap_elements(p2)</code></pre>
<p><img src="figure/estimateCellProportions.Rmd/unnamed-chunk-24-1.png" width="768" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-24-1">
Past versions of unnamed-chunk-24-1.png
</button>
</p>
<div id="fig-unnamed-chunk-24-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://atlassian.petermac.org.au/bitbucket/scm/os/paed-bal-meth-ref/blob/10147c01aaa38682abaf4f3694d1ce55fc634e06/docs/figure/estimateCellProportions.Rmd/unnamed-chunk-24-1.png" target="_blank">10147c0</a>
</td>
<td>
Jovana Maksimovic
</td>
<td>
2021-09-27
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.0.2 (2020-06-22)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS:   /config/binaries/R/4.0.2/lib64/R/lib/libRblas.so
LAPACK: /config/binaries/R/4.0.2/lib64/R/lib/libRlapack.so

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats4    parallel  stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] RColorBrewer_1.1-2                                 
 [2] patchwork_1.1.1                                    
 [3] forcats_0.5.1                                      
 [4] stringr_1.4.0                                      
 [5] dplyr_1.0.4                                        
 [6] purrr_0.3.4                                        
 [7] readr_1.4.0                                        
 [8] tidyr_1.1.2                                        
 [9] tibble_3.1.2                                       
[10] tidyverse_1.3.0                                    
[11] reshape2_1.4.4                                     
[12] ggplot2_3.3.5                                      
[13] FlowSorted.Blood.EPIC_1.8.0                        
[14] ExperimentHub_1.16.0                               
[15] AnnotationHub_2.22.0                               
[16] BiocFileCache_1.14.0                               
[17] dbplyr_2.1.0                                       
[18] nlme_3.1-152                                       
[19] quadprog_1.5-8                                     
[20] genefilter_1.72.1                                  
[21] IlluminaHumanMethylationEPICmanifest_0.3.0         
[22] IlluminaHumanMethylationEPICanno.ilm10b4.hg19_0.6.0
[23] minfi_1.36.0                                       
[24] bumphunter_1.32.0                                  
[25] locfit_1.5-9.4                                     
[26] iterators_1.0.13                                   
[27] foreach_1.5.1                                      
[28] Biostrings_2.58.0                                  
[29] XVector_0.30.0                                     
[30] SummarizedExperiment_1.20.0                        
[31] Biobase_2.50.0                                     
[32] MatrixGenerics_1.2.1                               
[33] matrixStats_0.59.0                                 
[34] GenomicRanges_1.42.0                               
[35] GenomeInfoDb_1.26.7                                
[36] IRanges_2.24.1                                     
[37] S4Vectors_0.28.1                                   
[38] BiocGenerics_0.36.1                                
[39] limma_3.46.0                                       
[40] glue_1.4.2                                         
[41] here_1.0.1                                         
[42] workflowr_1.6.2                                    

loaded via a namespace (and not attached):
  [1] readxl_1.3.1                  backports_1.2.1              
  [3] NMF_0.23.0                    plyr_1.8.6                   
  [5] splines_4.0.2                 BiocParallel_1.24.1          
  [7] gridBase_0.4-7                digest_0.6.27                
  [9] htmltools_0.5.1.1             fansi_0.5.0                  
 [11] magrittr_2.0.1                memoise_2.0.0.9000           
 [13] doParallel_1.0.16             cluster_2.1.0                
 [15] annotate_1.68.0               modelr_0.1.8                 
 [17] askpass_1.1                   siggenes_1.64.0              
 [19] prettyunits_1.1.1             colorspace_2.0-2             
 [21] rvest_0.3.6                   blob_1.2.1                   
 [23] rappdirs_0.3.3                haven_2.3.1                  
 [25] xfun_0.23                     jsonlite_1.7.2               
 [27] crayon_1.4.1                  RCurl_1.98-1.3               
 [29] GEOquery_2.58.0               survival_3.2-7               
 [31] registry_0.5-1                gtable_0.3.0                 
 [33] zlibbioc_1.36.0               DelayedArray_0.16.3          
 [35] Rhdf5lib_1.12.1               HDF5Array_1.18.1             
 [37] scales_1.1.1                  DBI_1.1.1                    
 [39] rngtools_1.5                  Rcpp_1.0.6                   
 [41] xtable_1.8-4                  progress_1.2.2               
 [43] bit_4.0.4                     mclust_5.4.7                 
 [45] preprocessCore_1.52.1         httr_1.4.2                   
 [47] ellipsis_0.3.2                farver_2.1.0                 
 [49] pkgconfig_2.0.3               reshape_0.8.8                
 [51] XML_3.99-0.5                  utf8_1.2.1                   
 [53] labeling_0.4.2                tidyselect_1.1.0             
 [55] rlang_0.4.11                  later_1.1.0.1                
 [57] AnnotationDbi_1.52.0          cellranger_1.1.0             
 [59] munsell_0.5.0                 BiocVersion_3.12.0           
 [61] tools_4.0.2                   cachem_1.0.4                 
 [63] cli_3.0.0                     generics_0.1.0               
 [65] RSQLite_2.2.5                 broom_0.7.4                  
 [67] evaluate_0.14                 fastmap_1.1.0                
 [69] yaml_2.2.1                    knitr_1.31                   
 [71] bit64_4.0.5                   fs_1.5.0                     
 [73] beanplot_1.2                  scrime_1.3.5                 
 [75] doRNG_1.8.2                   sparseMatrixStats_1.2.0      
 [77] whisker_0.4                   mime_0.10                    
 [79] nor1mix_1.3-0                 xml2_1.3.2                   
 [81] biomaRt_2.46.3                rstudioapi_0.13              
 [83] compiler_4.0.2                curl_4.3                     
 [85] interactiveDisplayBase_1.28.0 reprex_1.0.0                 
 [87] stringi_1.5.3                 highr_0.8                    
 [89] GenomicFeatures_1.42.1        lattice_0.20-41              
 [91] Matrix_1.3-2                  multtest_2.46.0              
 [93] vctrs_0.3.8                   pillar_1.6.1                 
 [95] lifecycle_1.0.0               rhdf5filters_1.2.0           
 [97] BiocManager_1.30.10           data.table_1.13.6            
 [99] bitops_1.0-7                  httpuv_1.5.5                 
[101] rtracklayer_1.50.0            R6_2.5.0                     
[103] promises_1.2.0.1              codetools_0.2-18             
[105] MASS_7.3-53.1                 assertthat_0.2.1             
[107] rhdf5_2.34.0                  pkgmaker_0.32.2              
[109] openssl_1.4.3                 rprojroot_2.0.2              
[111] withr_2.4.2                   GenomicAlignments_1.26.0     
[113] Rsamtools_2.6.0               GenomeInfoDbData_1.2.4       
[115] hms_1.0.0                     grid_4.0.2                   
[117] base64_2.0                    rmarkdown_2.6                
[119] DelayedMatrixStats_1.12.3     illuminaio_0.32.0            
[121] git2r_0.28.0                  lubridate_1.7.9.2            
[123] shiny_1.6.0                  </code></pre>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
