---
title: "Methylation profiling of paediatric Bronchoalveolar Lavage (BAL)"
subtitle: "Establishing a methylation reference panel of BAL consitutent cell types"
author: "Jovana Maksimovic"
date: "`r Sys.Date()`"
output:
  pdf_document: default
  html_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# Data import

Load all the packages required for analysis.

```{r, message=FALSE, warning=FALSE}
library(here)
library(workflowr)
library(limma)
library(minfi)
library(missMethyl)
library(matrixStats)
library(minfiData)
library(stringr)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
library(IlluminaHumanMethylationEPICmanifest)
library(tidyverse)
library(ggplot2)
library(patchwork)
library(glue)
```

Load the EPIC array annotation data that describes the genomic context of each of the probes on the array.

```{r}
# Get the EPICarray annotation data
annEPIC <- getAnnotation(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
annEPIC %>% data.frame %>%
  dplyr::select("chr",
                "pos",
                "strand",
                "UCSC_RefGene_Name",
                "UCSC_RefGene_Group") %>%
  head() %>% knitr::kable()
```

Read the sample information and IDAT file paths into R. 

```{r, message=FALSE}
# absolute path to the directory where the data is (relative to the Rstudio project)
dataDirectory <- here("data")

# read in the sample sheet for the experiment
read.metharray.sheet(dataDirectory, 
                     pattern = "Samplesheet_BAL_reference.csv") %>%
  mutate(Basename = gsub("c(\"","", Basename, fixed=TRUE)) %>%
  mutate(Sample_ID = paste(Slide, Array, sep = "_")) %>%
  mutate_if(is.character, stringr::str_replace_all, pattern = "Granuloycte", 
            replacement = "Granulocyte") %>%
  mutate_if(is.character, stringr::str_replace_all, pattern = "Epithelialcell", 
            replacement = "EpithelialCell") %>%
  mutate_if(is.character, stringr::str_replace_all, pattern = "Old", 
            replacement = "Erasmus MC") %>%
  mutate_if(is.character, stringr::str_replace_all, pattern = "New", 
            replacement = "GenomeScan") %>%
  dplyr::select(Sample_ID, Sample_Group, Sample_source, Sample_run, 
                Basename, Num_individuals) -> targets

targets %>% dplyr::select(-Basename) %>% knitr::kable()
```

Read in the raw methylation data.

```{r}
# read in the raw data from the IDAT files
rgSet <- read.metharray.exp(targets = targets)
rgSet
```

# Quality control

Calculate the detection P-values for each probe so that we can check for any failed samples.

```{r}
# calculate the detection p-values
detP <- detectionP(rgSet)
head(detP[, 1:5]) %>% knitr::kable()
```

We can see the mean detection p-values across all the samples for the two runs. 

```{r, fig.height=8, fig.width=7}
dat <- reshape2::melt(colMeans(detP))
dat$type <- targets$Sample_Group
dat$sample <- targets$Sample_source
dat$run <- targets$Sample_run

pal <- scales::hue_pal()(length(unique(targets$Sample_Group)))
names(pal) <- c("Granulocyte","Raw","Lymphocyte",
                "Macrophage","EpithelialCell")
  
# examine mean detection p-values across all samples to identify failed samples
p <- ggplot(dat, aes(x = sample, y = value, fill = type)) +
  geom_bar(stat = "identity") +
  facet_wrap(vars(run), ncol = 2, scales = "free_x") +
  labs(x = "Sample", y = "Detection p-value", fill = "Sample type") +
  theme(axis.text.y = element_text(size = 7),
        axis.text.x = element_text(angle = 45, hjust = 1),
        legend.position = "bottom") +
  scale_fill_manual(values = pal)
p
```
## Sample filtering

Filter out samples with mean detection p-values > 0.01 to avoid filtering out too many probes downstream.

```{r}
keep <- colMeans(detP) < 0.01
table(keep)

rgSet <- rgSet[, keep]
targets <- targets[keep, ]
detP <- detP[, keep]
```

## Normalisation

Normalise the data.

```{r}
# normalize the data; this results in a GenomicRatioSet object
mSetSq <- preprocessQuantile(rgSet)
mSetSq

# create a MethylSet object from the raw data for plotting
mSetRaw <- preprocessRaw(rgSet)
```

```{r}
bSq <- getBeta(mSetSq)
bRaw <- getBeta(mSetRaw)

sq <- reshape2::melt(bSq)
sq$type <- targets$Sample_Group
sq$process <- "Normalised"

raw <- reshape2::melt(bRaw)
raw$type <- targets$Sample_Group
raw$process <- "Raw"

dat <- bind_rows(sq, raw)

colnames(dat)[1:3] <- c("cpg", "ID", "beta")

ggplot(data = dat, 
       aes(x = beta, colour = type)) +
  geom_density(show.legend = NA) +
  labs(colour = "Sample type") +
  facet_wrap(vars(process), ncol = 2) +
  scale_color_manual(values = pal)
```

## Data exploration (before probe filtering)

Explore the data to look for any structure. As expected, sex is the most significant source of variation. The patient samples, in particular, are clearly grouping by sex.   

```{r, fig.asp=1}
mDat <- getM(mSetSq)
mds <- plotMDS(mDat, top = 1000, gene.selection="common", plot = FALSE)
dat <- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group, 
              run = targets$Sample_run,
              source = targets$Sample_source,
              sex = targets$Sex)

p1 <- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = "Sample type", shape = "Run",
       x = "Principal component 1", 
       y = "Principal component 2") +
  ggtitle("All data") +
  scale_color_manual(values = pal)

p1
```

Colour ONLY the sorted cells samples using different variables. Encouragingly, the cell types cluster tightly when samples from both runs are combined, indicating that differences between cell types are more significant than any bath effects between the two runs. The cell types also cluster as expected for each of the individual runs. 

```{r, fig.asp=1}
cells <- !(targets$Sample_Group %in% c("Raw"))
mds <- plotMDS(mDat[, cells], top = 1000, gene.selection="common", plot = FALSE)
dat <- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group[cells], 
              run = targets$Sample_run[cells],
              source = targets$Sample_source[cells])

p1 <- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = "Sample type", shape = "Run",
       x = "Principal component 1", 
       y = "Principal component 2") +
  ggtitle("Cell types") +
  scale_color_manual(values = pal[-2])

p1
```

## Probe filtering

Filter out poor performing probes, sex chromosome probes, SNP probes and cross reactive probes. 

```{r}
# ensure probes are in the same order in the mSetSq and detP objects
detP <- detP[match(featureNames(mSetSq), rownames(detP)),]
# remove any probes that have failed in one or more samples
keep <- rowSums(detP < 0.01) == ncol(mSetSq)
# subset data
mSetSqFlt <- mSetSq[keep,]
mSetSqFlt
```

Calculate M and beta values for downstream use in analysis and visulalisation.

```{r}
# calculate M-values and beta values for downstream analysis and visualisation
mVals <- getM(mSetSqFlt)
head(mVals[,1:5]) %>% knitr::kable()
```

```{r}
bVals <- getBeta(mSetSqFlt)
head(bVals[,1:5]) %>% knitr::kable()
```

Remove SNP probes and multi-mapping probes.

```{r, message=FALSE, warning=FALSE}
mVals <- DMRcate::rmSNPandCH(mVals, mafcut = 0, rmcrosshyb = TRUE, rmXY = FALSE)
bVals <- DMRcate::rmSNPandCH(bVals, mafcut = 0, rmcrosshyb = TRUE, rmXY = FALSE)
dim(mVals)
```

Remove sex chromosome probes.

```{r, message=FALSE, warning=FALSE}
mValsNoXY <- DMRcate::rmSNPandCH(mVals, mafcut = 0, rmcrosshyb = TRUE, 
                                 rmXY = TRUE)
bValsNoXY <- DMRcate::rmSNPandCH(bVals, mafcut = 0, rmcrosshyb = TRUE, 
                                 rmXY = TRUE)
dim(mValsNoXY)
```

## Data exploration (after probe filtering)

After filtering, sorted cell types still cluster tightly together. The mixed samples are centered between the sorted cell types, which is indicative of the mixtures.

### **Supplementary Figure 5**

```{r, fig.asp=1}
mDat <- mValsNoXY
mds <- plotMDS(mDat, top = 1000, gene.selection="common", plot = FALSE)
dat <- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group, 
              run = targets$Sample_run,
              source = targets$Sample_source,
              sex = targets$Sex)

p1 <- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = "Sample type", shape = "Run",
       x = "Principal component 1", 
       y = "Principal component 2") +
  ggtitle("All data") +
  scale_color_manual(values = pal)

p1
```

After filtering, the sorted cell types cluster tightly together and there is no evidence of a significant batch effect.

```{r, fig.asp=1}
cells <- !(targets$Sample_Group %in% c("Raw"))
mds <- plotMDS(mDat[, cells], top = 1000, gene.selection="common", plot = FALSE)
dat <- tibble(x = mds$x, 
              y = mds$y, 
              sample = targets$Sample_Group[cells], 
              run = targets$Sample_run[cells],
              source = targets$Sample_source[cells])

p1 <- ggplot(dat, aes(x = x, y = y, colour = sample)) +
  geom_point(aes(shape = run), size = 3) +
  labs(colour = "Sample type", shape = "Run",
       x = "Principal component 1", 
       y = "Principal component 2") +
  ggtitle("Cell types") +
  scale_color_manual(values = pal[-2])

p1
```

Calculate the percentage of variance explained by principal components.

```{r}
PCs <- prcomp(t(mValsNoXY), scale = TRUE)
summary(PCs)
```
### **Supplementary Figure 2**

```{r}
summary(PCs)$importance %>% 
  t() %>%
  data.frame() %>%
  rownames_to_column(var = "PC") %>%
  slice_head(n = 10) %>%
  ggplot(aes(x = forcats::fct_reorder(PC, 1:10), 
             y = Proportion.of.Variance * 100,
             group = 1)) +
  geom_point(size = 4) +
  geom_line() +
  labs(y = "% Variance explained",
       x = "Principal component") +
  cowplot::theme_cowplot()
```

### **Figure 2**

```{r, fig.width=9, fig.height=5}
dims <- list(c(1,2), c(1,3), c(2,3))
pcvar <- summary(PCs)$importance
p <- vector("list", length(dims))

for(i in 1:length(p)){
  mds <- plotMDS(mValsNoXY[, cells], top = 1000, gene.selection = "common", 
                 plot = FALSE, dim.plot = dims[[i]])
  dat <- tibble(x = mds$x, 
                y = mds$y, 
                sample = targets$Sample_Group[cells], 
                run = targets$Sample_run[cells],
                source = targets$Sample_source[cells])
  
  p[[i]] <- ggplot(dat, aes(x = x, y = y, colour = sample)) +
    geom_point(aes(shape = run), size = 3) +
    labs(colour = "Sample type", shape = "Run",
         x = glue("Principal component {dims[[i]][1]} ({round(pcvar[2,dims[[i]][1]]*100,1)}%)"), 
         y = glue("Principal component {dims[[i]][2]} ({round(pcvar[2,dims[[i]][2]]*100,1)}%)")) +
    scale_color_manual(values = pal[-2]) + 
    theme(legend.direction = "vertical")
  
}

(p[[1]] | p[[2]] | p[[3]]) + plot_layout(guides = "collect") &
  theme(legend.position = "bottom")
```

## Assess raw BAL for presence of additional cell types

If the 4 sorted cell types are fully representative of raw BAL, then fully methylated/unmethylated loci across all the sorted cell types should also be fully methylated/unmethylated in raw BAL. We identified 3479 probes with mean beta values either greater than 0.95 or less than 0.05 across all the sorted cell types. These probes are consistently fully methylated or fully unmethylated across both the sorted cell types and the raw BAL samples, confirming that the raw BAL is unlikely to contain additional nucleated cell types. 

### **Supplementary Figure 4**

```{r, fig.asp=1}
cellTypes <- unique(targets$Sample_Group)[-5]
tmp <- bValsNoXY[, targets$Sample_Group %in% cellTypes]
keep <- which(rowMeans(tmp) >= 0.95 | rowMeans(tmp) <= 0.05)
o <- order(targets$Sample_Group)

NMF::aheatmap(bValsNoXY[keep, o],
         annCol = list(SampleType = as.character(targets$Sample_Group[o])), 
         labCol = NA, labRow = NA, annColors = list(pal),
         Colv = NA)
```

## Pooling strategy assessment

We assessed the impact of our pooling strategy on the methylomes of the sorted cell types by examining the variance of pools with different numbers of contributing donors. If the variance of beta values (or M values) in the methylomes derived from pools with fewer donors is not greater than the variance of methylomes of pools from a higher number of donors, then our pooling strategy is unlikely to be significantly biasing the methylation data. 
Thus, we examined the relationship between the number of individuals contributing to a pooled sample and the variance across CpGs in the sample. The result indicates that there is no statistically significant relationship between the number of individuals contributing to a pool and its variance across CpGs. 

### **Supplementary Figure 1**

```{r}
data.frame(poolVars = colVars(mValsNoXY)) %>%
  mutate(cellType = targets$Sample_Group,
         num = targets$Num_individuals) %>%
  dplyr::filter(cellType != "Raw" ) %>%
  mutate(cellType = forcats::fct_drop(cellType)) -> dat

f <- lm(poolVars ~ num, data = dat)
s <- summary(f)

df <- data.frame(label = as.character(glue::glue("Adj. R^2 = {signif(s$adj.r.squared, 4)}\n
                         Intercept = {signif(f$coef[[1]], 4)}\n
                         Slope = {signif(f$coef[[2]], 4)}\n
                         P-value = {signif(s$coef[2, 4], 4)}")),
                 x = 9,
                 y = 6.75,
                 hjust = 0,
                 vjust = 1,
                 col = "black")

ggplot(dat, aes(x = num, y = poolVars, color = cellType)) +
  geom_point() +
  geom_abline(slope = coef(f)[[2]], intercept = coef(f)[[1]],
              linetype = "dashed", colour = "red") +
  scale_color_manual(values = pal[-2]) +
  scale_x_continuous(breaks = 1:nrow(dat)) +
  cowplot::theme_cowplot(font_size = 12) +
  labs(x = "No. contributing individuals", 
       y = "Variance across CpGs", 
       color = "Sample type") +
  ggtext::geom_richtext(data = df, inherit.aes = FALSE,
                        aes(x, y, label = label, hjust = hjust, 
                            vjust = vjust),
                        fill = NA, label.color = NA, 
                        size = 3, color = "black")
```

# Save processed data

The data is of good quality and shows no evidence of unexpected sources of variation. 
Save the various data objects for faster downstream analysis.

```{r}
outFile <- here("data/processedData.RData")
if(!file.exists(outFile)){
  save(annEPIC, mSetSqFlt, rgSet, mVals, bVals, targets, mValsNoXY, bValsNoXY, 
       pal, cells, file = outFile)
}
```


